test/core/base/virtual-node/abstract-virtual-node.js

```diff
@@ -14,6 +14,17 @@ describe('AbstractVirtualNode', function() {
     assert.throws(fn);
   });
 
+  it('should throw an error when accessing attrNames', function() {
+    function fn() {
+      var abstractNode = new axe.AbstractVirtualNode();
+      if (abstractNode.attrNames.length) {
+        return;
+      }
+    }
+
+    assert.throws(fn);
+  });
```

Comment:
This test seems to be testing that JavaScript property access works as expected. Is it any different than doing this:

```
function fn(){
  var a = {}
  a.b.c.d.e.f.g.h.length
  return 7
}

assert.throws(fn)
```

Reply:
Sorry I don't quite understand. Are you saying you'd prefer not having the `.length` part or that the test itself isn't useful?

Reply:
Assuming I'm following the code correctly, the `attrNames` getter returns nothing (`undefined`), so doing `undefined.length` obviously throws. I think asserting that the exception raised by `get attrNames(){}` matches what we expect. For example:

```suggestion
  it('should throw an error when accessing attrNames', function() {
    function fn() {
      var abstractNode = new axe.AbstractVirtualNode();
      return abstractNode.attrNames;
    }

    assert.throws(fn, 'VirtualNode class must have an "attrNames" property');
  });
```

Reply:
This is not a "big deal", nor a blocker. Feel free to merge since Wilco has already approved 👍

Reply:
Its good. I agree that with the `.length` part it may not through because of the reason we expect.